Use REST interface for AppVeyor

ngbrown opened the pull request:
The use of a blocking command-line tool per test is quite slow.  It caused my tests to be canceled at the 60 minute limit with only 2,559 completed.

The changes use a Async queue and HTTP batch POST to minimize the time impact on builds with a large number of tests.  With these changes, all 5,781 test complete in 19 minutes.

Faizan2304 commented:
@ngbrown : Thank you for your contribution 👍. I will take a look on this weekend.

Faizan2304 commented on src/Appveyor.TestLogger/AppveyorApiTests.cs:
```diff
@@ -0,0 +1,15 @@
+namespace Microsoft.VisualStudio.TestPlatform.Extensions.Appveyor.TestLogger
+{
+	internal class AppveyorApiTests
```
Do we need this class?

ngbrown replied:
Not needed since I'm building JSON manually instead of using a library. - Removed.

Faizan2304 commented on src/Appveyor.TestLogger/AppveyorLogger.cs:
```diff
@@ -24,14 +23,26 @@ class AppveyorLogger : ITestLogger
-        /// <param name="events">Events that can be registered for.</param>
-        /// <param name="testRunDirectory">Test Run Directory</param>
-        public void Initialize(TestLoggerEvents events, string testRunDirectory)
+	    private AppveyorLoggerQueue queue;
```
indentation

ngbrown replied:
I had accidentally added a mess of tab characters and spaces.  Added `.editorconfig` to enforce standardization.

Faizan2304 commented on src/Appveyor.TestLogger/AppveyorLogger.cs:
```diff
@@ -24,14 +23,26 @@ class AppveyorLogger : ITestLogger
+		/// </summary>
+		/// <param name="events">Events that can be registered for.</param>
+		/// <param name="testRunDirectory">Test Run Directory</param>
+		public void Initialize(TestLoggerEvents events, string testRunDirectory)
```
indentation

ngbrown replied:
fixed

Faizan2304 commented on src/Appveyor.TestLogger/AppveyorLogger.cs:
```diff
@@ -67,18 +78,19 @@ private void TestMessageHandler(object sender, TestRunMessageEventArgs e)
 
-            allArgs.Add("-Name " + name);
-            allArgs.Add("-Framework " + e.Result.TestCase.ExecutorUri.ToString());
+			var testResult = new Dictionary<string, string>();
```
please fix the indentation of this file

ngbrown replied:
fixed

Faizan2304 commented on src/Appveyor.TestLogger/AppveyorLoggerQueue.cs:
```diff
@@ -0,0 +1,73 @@
+			queue.Cancel();
+			try
+			{
+				consumeTask.Wait(TimeSpan.FromSeconds(10));
```
I was just wondering if there is any particular reason behind choosing 10 sec.

ngbrown replied:
Not specifically.  The API calls are to a `localhost` and I expected them to be fast.  Just thought there should be something.  Do you have a suggestion for a better length?

ngbrown commented:
@Faizan2304 I found an issue with how I was handling the Aync at the `Flush` point.

I also fixed up the build script quite a bit to perform two-phase compilation so that the NuGet packages just built will be the ones tested.  The `TastResult.xml` or `loggerFile.xml` are deleted before test and then if they aren't created the AppVeyor build should fail.

I've rebased everything for a clean set of commits.
